modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java

```diff
@@ -159,6 +160,15 @@ public static void send(
         String preserveAddressingProperty = (String) synapseOutMessageContext.getProperty(
                 SynapseConstants.PRESERVE_WS_ADDRESSING);
         MessageContext axisOutMsgCtx = cloneForSend(originalInMsgCtx, preserveAddressingProperty);
+
+        // set "INTERNAL_TRANSACTION_COUNTED" property in axisOutMsgCtx's message context, if it is present in the message
+        // context of the originalInMsgCtx
+        Object transactionProperty = originalInMsgCtx.getProperty(BaseConstants.INTERNAL_TRANSACTION_COUNTED);
+        if (transactionProperty instanceof Boolean) {
```

Comment:
Since this is an internally set property, do we really need to check instanceof?

Reply:
I think we don't need it. But we need to do a null check if we're removing this

Reply:
null check and Boolean.valueOf should work.